src/preferences/dialog/dlgprefcolors.cpp

```diff
@@ -227,18 +217,62 @@ void DlgPrefColors::slotHotcuePaletteChanged(const QString& paletteName) {
     comboBoxHotcueDefaultColor->setCurrentIndex(defaultColor);
 }
 
-void DlgPrefColors::slotEditClicked() {
-    pushButtonEdit->hide();
-    labelCustomPalette->hide();
-    widgetSpacer->hide();
-    groupBoxPaletteEditor->show();
+void DlgPrefColors::slotEditTrackPaletteClicked() {
+    QString trackColorPaletteName = comboBoxTrackColors->currentText();
+    openColorPaletteEditor(trackColorPaletteName, false);
 }
 
-void DlgPrefColors::slotCloseClicked() {
-    groupBoxPaletteEditor->hide();
-    widgetSpacer->show();
-    pushButtonEdit->show();
-    labelCustomPalette->show();
+void DlgPrefColors::slotEditHotcuePaletteClicked() {
+    QString hotcueColorPaletteName = comboBoxHotcueColors->currentText();
+    openColorPaletteEditor(hotcueColorPaletteName, true);
+}
+
+void DlgPrefColors::openColorPaletteEditor(
+        const QString& paletteName,
+        bool editHotcuePalette) {
+    std::unique_ptr<ColorPaletteEditor> pColorPaletteEditor =
+            std::make_unique<ColorPaletteEditor>(this);
+
+    if (editHotcuePalette) {
+        connect(pColorPaletteEditor.get(),
+                &ColorPaletteEditor::paletteChanged,
+                this,
+                &DlgPrefColors::hotcuePaletteUpdated);
+    } else {
+        connect(pColorPaletteEditor.get(),
+                &ColorPaletteEditor::paletteChanged,
+                this,
+                &DlgPrefColors::trackPaletteUpdated);
+    }
+    connect(pColorPaletteEditor.get(),
+            &ColorPaletteEditor::paletteRemoved,
+            this,
+            &DlgPrefColors::palettesUpdated);
+
+    pColorPaletteEditor->initialize(m_pConfig, paletteName);
+    pColorPaletteEditor->exec();
+}
+
+void DlgPrefColors::trackPaletteUpdated(const QString& trackColors) {
+    QString hotcueColors = comboBoxHotcueColors->currentText();
+    int defaultColor = comboBoxHotcueDefaultColor->currentIndex();
+
+    loadSettings();
+
+    comboBoxHotcueColors->setCurrentText(hotcueColors);
+    comboBoxTrackColors->setCurrentText(trackColors);
+    comboBoxHotcueDefaultColor->setCurrentIndex(defaultColor);
+}
+
+void DlgPrefColors::hotcuePaletteUpdated(const QString& hotcueColors) {
+    QString trackColors = comboBoxTrackColors->currentText();
+    int defaultColor = comboBoxHotcueDefaultColor->currentIndex();
+
+    loadSettings();
+
+    comboBoxHotcueColors->setCurrentText(hotcueColors);
+    comboBoxTrackColors->setCurrentText(trackColors);
+    comboBoxHotcueDefaultColor->setCurrentIndex(defaultColor);
```

Comment:
```suggestion
    // Select "By hotcue number". A hotcue index in one palette has no meaning in another palette.
    comboBoxHotcueDefaultColor->setCurrentIndex(0);
```

Reply:
It has, it has. If you wish to use the first index as default or if you have selected the "color by number" setting. 
This change was requested by @Holzhaus